packages/bff/src/services/agreementService.ts

```diff
@@ -1,40 +1,80 @@
 /* eslint-disable functional/immutable-data */
+/* eslint-disable max-params */
 import { agreementApi, catalogApi } from "pagopa-interop-api-clients";
 import { AgreementProcessClient } from "../providers/clientProvider.js";
 import { Headers } from "../utilities/context.js";
 
+// Fetched all agreements in a recursive way
+export const getAllAgreements = async (
+  agreementProcessClient: AgreementProcessClient,
+  headers: Headers,
+  consumerIds: string[] = [],
+  eserviceIds: string[] = [],
+  producerIds: string[] = [],
+  states: agreementApi.AgreementState[] = [],
+  start: number = 0
+): Promise<agreementApi.Agreement[]> => {
+  const agreements = (
+    await getAgreementsFrom(
+      agreementProcessClient,
+      headers,
+      start,
+      consumerIds,
+      eserviceIds,
+      producerIds
+    )
+  ).results;
+
+  if (agreements.length >= 50) {
+    return agreements.concat(
+      await getAllAgreements(
+        agreementProcessClient,
+        headers,
+        consumerIds,
+        eserviceIds,
+        producerIds,
+        states,
+        start + 50
+      )
+    );
+  }
+  return agreements;
```

Comment:
Eric did implement [here](https://github.com/pagopa/interop-be-monorepo/pull/764/files#diff-8e30f6ae0147b4b664468b1e29476bd48cb49a7928b5ed7bdf2fd5ac9653abdbR3-R16) an handy function that does this in the commons, I would wait for it to be merged in order to also implement it here. WDYT?